### PR TITLE
BattleShip-14/UI-6/BattleShip-10 Style for Delete Account button. Restore user. Update user info after log in.

### PR DESCRIPTION
**Pull request by kolokolochek**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -8,12 +8,28 @@ class User < ApplicationRecord
...
+    find_and_update_user(params) || create(params)
   end
 
   private
 
   def set_nick
     self.nick = name
   end
+
+  def self.find_and_update_user(params)
```
Comment (IlyaUmanets) [inline app/models/user.rb]: this method do a lot of things. try to put empty lines to divide this

File: app/views/profiles/show.slim [hunk 2]
```diff
@@ -9,5 +9,8 @@
       li
         h2= I18n.t('page.profile.title.stat')
         = render 'stat'
-    .delete-profile-wrapper
-      = link_to 'delete', profile_path, method: :delete, confirm: 'Are you sure?'
+      li
+        h2= I18n.t('page.profile.title.delete')
+        .delete-profile
+          p= I18n.t('notice.delete_account')
+          = link_to I18n.t('button.delete_account'), profile_path, method: :delete, data: { confirm: I18n.t('notice.confirm_delete_account'), disable_with: I18n.t('button.disable_with') }
```
Comment (IlyaUmanets) [inline app/views/profiles/show.slim]: too long line

File: spec/controllers/sessions_controller_spec.rb [hunk 3]
```diff
@@ -44,6 +44,30 @@
...
+        create_request
+
+        expect(current_user.reload.deleted?).to be false
+        expect(current_user.name).to eq('kolokol')
+      end
+
+      context 'when user updated social account' do
+        before { auth_hash_response_for(current_user) }
+
+        it 'should update name and image' do
```
Comment (IlyaUmanets) [inline spec/controllers/sessions_controller_spec.rb]: just `it 'updates....'`

Comment (kolokolochek) [reply spec/controllers/sessions_controller_spec.rb]: I wrote 'should update' b/c I will see next message for error spec:
```
#create when user updated social account should update name and image
```
^ for me it more correct than next message:
```
#create when user updated social account updates name and image
```
@IlyaUmanets , does it make sence?

**Pull request merged**